pkg/template/evaluation_ctx.go

```diff
@@ -156,9 +157,15 @@ func (e *EvaluationCtx) TplStartNodeAnnotation(
 	if err != nil {
 		return starlark.None, err
 	}
-
 	annName := AnnotationName(annNameStr)
-	annVals := args.Index(2).(starlark.Tuple)
+
+	lineNum, err := strconv.Atoi(args.Index(2).String())
+	if err != nil {
+		//what error would this be? string should be line number... when would it not
+		return starlark.None, err
```

Comment:
Looking at this error case got me thinking...

It's technically possible that the line number of the annotation is fully unknown. This can happen, for example, when an AST is being built in memory. In that case, there _is no_ comment to refer to.

Perhaps an appropriate thing to do here is...

- on the calling side:
    - if the annotation's Position is known, pass it along as you are.
    - if not, pass in `None`.
- on the receiving end...
   - if the annotation's Position is known, then Bob's yer uncle (and if for some crazy reason that's not an integer... seems like panic'ing is in order).
   - if the value is `None`, then build the "source" from the annotation, itself (e.g. `NodeAnnotation.AsSourceString()`)

Happy to chat more about this... feels like something that _could_ be deferred so long as we have a simple fallback strategy for when positions are unknown.